### ISSUE TITLE
setarch->CARCH mapping

### ISSUE THREAD

**Issue by deep-42-thought** [archlinux/devtools#44]
It might be desirable to provide `setarch` with the same architecture for differing `CARCH`s. E.g. archlinux32 has "i686" and "pentium4" architectures, which differ in config files and `CFLAGS` and possibly also inside the `PKGBUILD`s (e.g. different patches might be necessary). Currently, we do this via a hard-coded mapping inside `arch-nspawn`: https://git.archlinux32.org/devtools32/commit/?id=45513b8260331f86418384c2b5df191cc62e2f6d
but as discussed on irc, it might be desirable to export this into a separate config file inside `/usr/share/devtools`, which would enable archlinux and downstream distributions to add even more such pseudo-architectures to build packages optimized for more specific hardware.
If that is something, that would be desired, I can look into how to convert our hack into such a solution and open a pull-request for it.

**Comment by petronny** [comment]
There is another approach.
We can hack `setarch.c` to have the CARCH -> kernel arch mapping.

For example,
```
sed -i '140a{PER_LINUX32, "armv6h", "armv7l"},' sys-utils/setarch.c
sed -i '140a{PER_LINUX32, "armv7h", "armv7l"},' sys-utils/setarch.c
```
this will work for both armv6h and armv7h on an armv7l kernel.

I think there is a similar approach for aarch64.

**Comment by anthraxx** [comment]
fixed via 30ed6920c7e7962b7161bc2405c63e9a1ab95be2
